syntax = "proto3";
package protos.connectors.assets;

import "google/protobuf/wrappers.proto";
import "protos/base.proto";
import "protos/connectors/connector.proto";

message SshServerAssetModel {
  google.protobuf.StringValue name = 1;
}

message SshServerAssetOptions {
  repeated string ssh_servers = 1;
}

message RemoteServerAssetModel {
  google.protobuf.UInt64Value id = 1;
  Source connector_type = 2;
  SourceModelType type = 3;
  sfixed64 last_updated = 4;
  oneof asset {
    SshServerAssetModel ssh_server = 5;
  }
}

message RemoteServerAssets {
  repeated RemoteServerAssetModel assets = 1;
}
